vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -173,10 +173,10 @@ public void close() {
     requireNonNull(resultHandler, "resultHandler cannot be null");
 
     boolean id = query.containsKey(ID_FIELD);
-    query = encodeKeyWhenUseObjectId(query);
+    JsonObject encodedQuery = encodeKeyWhenUseObjectId(query);
 
```

Comment:
I understand the rationale for not assigning back to a parameter - but I'm not 100% sure it's safe to change this.

@johnoliver - when you get a chance can you check this is a safe alteration?